### ISSUE TITLE
Instructor's Assistant Permissions

### ISSUE THREAD

**Issue by nrherron92** [gocodebox/lifterlms#2122]
### Reproduction Steps

+ This isn't connected to a ticket I just found this while looking into the other instructor permissions. 
+ Create an instructor's assistant user
+ Assign them to an instructor who has created a course and is the author
+ Switch into that user

### Expected Behavior

+ they can edit courses that are authored by the instructors they’ve been assigned to

### Actual Behavior

+ They cannot edit the courses _unless_ they are assigned as an instructor.
+ However, if they are assigned as an instructor they have the ability to add/modify the access plans

### Error Messages / Logs

+ Include any relevant error messages or log files
```
<!-- Paste error logs / backtraces below this line -->

```

### System and Environment Information

<details>
<summary>System Report</summary>

```
Wordpress
-------------------------------------------

Home Url: https://natalie-test-site.myliftersite.com
Site Url: https://natalie-test-site.myliftersite.com
Login Url: https://natalie-test-site.myliftersite.com/wp-login.php
Version: 5.9.3
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_US
Multisite: No
Page For Posts: Not Set
Page On Front: Homepage (#1398) [https://natalie-test-site.myliftersite.com/]
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 6.3.0
Db Version: 6.3.0
Course Catalog: Course Catalog (#35) [https://natalie-test-site.myliftersite.com/courses/]
Membership Catalog: Membership Catalog (#36) [https://natalie-test-site.myliftersite.com/memberships/]
Student Dashboard: Dashboard (#38) [https://natalie-test-site.myliftersite.com/dashboard/]
Checkout Page: Checkout (#715) [https://natalie-test-site.myliftersite.com/checkout/]
Course Catalog Per Page: 9
Course Catalog Sorting: menu_order,ASC
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order
Site Membership: Not Set
Courses Endpoint: my-courses
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: no
Password Strength Meter: no
Minimum Password Strength: 
Terms Required: yes
Terms Page: Activate (#443) [https://natalie-test-site.myliftersite.com/activate/]
Checkout Names: 
Checkout Address: 
Checkout Phone: 
Checkout Email Confirmation: no
Open Registration: yes
Registration Names: 
Registration Address: 
Registration Phone: 
Registration Voucher: 
Registration Email Confirmation: no
Account Names: 
Account Address: 
Account Phone: 
Account Email Confirmation: no
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: no
Country: IT
Currency: USD
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: yes
Email From Address: team+sandbox@lifterlms.com
Email From Name: A LifterLMS Sandbox
Email Footer Text: 
Email Header Image: 
Cert Bg Width: 1123
Cert Bg Height: 794
Cert Legacy Compat: no


Constants
-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined
LLMS_REST_DISABLE: undefined
LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined
LLMS_SITE_IS_CLONE: undefined


Gateways
-------------------------------------------

Manual: Enabled
Manual Logging: no
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.7.37
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 5000
Php Max Upload Size: 512 MB
Php Memory Limit: 256M
Php Post Max Size: 1024M
Php Soap: Yes
Php Suhosin: No
Php Time Limt: 30
Php Version: 8.0.17
Software: Apache/2.4.53 (Unix) OpenSSL/1.1.1
Wp Memory Limit: 256M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.88 Safari/537.36


Theme
-------------------------------------------

Name: LifterLMS LaunchPad
Version: 2.4.9
Themeuri: https://lifterlms.com/launchpad/
Authoruri: https://gocodebox.com
Template: 
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

LifterLMS: 6.3.0
User Switching: 1.5.8


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No


Template Overrides
-------------------------------------------




```

</details>

This issue has be recreated:
+ [x] Locally
+ [x] On a trial site
+ [ ] On a production website
+ [x] With only LifterLMS and a default theme

### Browser, Device, and Operating System Information

+ Browser name and version
+ Operating System name and version
+ Device name and version (if applicable)

**Comment by eri-trabiccolo** [comment]
I think instructor's assistants should be able to edit all the posts the parent instructor is an instructor for.
Does this make sense?
If so, what about [this draft](https://github.com/gocodebox/lifterlms/pull/2123)? What do you think @gocodebox/engineering ?

Alternatively we can change the logic and allow assistants to edit whatever the parent instructors can edit, which is a bit different.

**Comment by pondermatic** [comment]
@eri-trabiccolo, I think your draft PR will work. I'm surprised that instructor assistants can't already edit courses that their parent instructor is an instructor of. (How many times can I use the word "instructor" in a sentence?)

**Comment by thomasplevy** [comment]
I am pretty certain that this is working as expected. Instructors Assistant's are supposed to be *added to a course* in order to be able to help with that course.

The reason here is that some instructors may have multiple assistants and each assistant is assigned to a specific course. If we provide blanket access to all content for all of the instructor's assistants then we end up creating a problem for anyone using this as it is today.

The existing functionality is working as designed. 

Problems?

1) Don't want the assistant to show on the instructor's list? Use the "hidden" option
2) #2121 (This is a bug, to be addresses separately and shouldn't be fixed here)
3) Have *too many courses* and adding the assistants to each one is too cumbersome? Welp... Sorry for now I guess. I don't see this as a valid reason to change existing functionality and risk breaking expected behavior for any user who is expecting it to work as is.

We could add an option for each assistant that provides that instructor with access to *all* courses the instructor has. This would probably be okay and solve issue 3 above pretty simply. 

I could see adding to the bulk edit interface to allow bulk-editing instructors. Ultimately I feel like this *might* be easier than adding an option per-instructor to allow them unlimited access to all of an instructor's courses.

**Comment by eri-trabiccolo** [comment]
I think what you say makes sense.
The documentation is wrong then:
https://lifterlms.com/docs/roles-and-capabilities/#instructors-assistant
> Instructor’s Assistants are similar to Instructors but they can only edit courses that are authored by the instructors they’ve been assigned to. 

right?

1. ok
2. yeah, that will be addressed in a fix for #2121
3. yeah, makes sense.

> I could see adding to the bulk edit interface to allow bulk-editing instructors. Ultimately I feel like this might be easier than adding an option per-instructor to allow them unlimited access to all of an instructor's courses.

Nice, like in the Future.

Let's stick on this then: let's close this, and the related PR, and open a new issue for the bulk-editing instructors, as nice to have.
